plutus-chain-index/README_marconi.md

```diff
@@ -0,0 +1,159 @@
+
+### Specifying Slots to Query
+
+Through the query interface, you can specify a range of slots to query or indicate that the query is for blocks deeper than a specified number of slots. As a result, you can use simple state management for applications that work only with immutable blocks. 
+
+### Querying Volatile Data
+
+If your application requires faster response times, you can query volatile data on the head of the chain; however, you must take care to invalidate cached data when rollbacks occur. 
+
+## API Layers
+
+The core Marconi APIs are defined in Haskell in order to support simple use cases where all the components run together in a single process. 
+
+If your use cases require multiple synchronized instances for load balancing or for supporting different indexers, Marconi is designed with alternative transport layers on top of the core API that support network streaming and RPC calls. 
+
+## What Differentiates Marconi
+
+* Developed by IOG
+* Built in Haskell
+
+### DB Sync
+
+Cardano DB-sync is IOG's current indexing solution, but it uses lots of memory and days/week to sync. DB-sync takes an index everything approach to put all information into a database that can support any query at the cost of requiring a large DB server to run it. 
+
+In contrast, Marconi requires the user to customize it for the specific application so that only relevant information is streamed or stored. Marconi will be a scalable solution that allows the dApp developer to index whatever is important for their unique use case. 
+
+### Ogmios
+
+Ogmios exposes the basic node interface as a web service. This is a great option for queries that can be made directly to the node for applications that want a web service interface.
+
+### Oura
+
+Oura provides very similar functionality with an implementation in Rust and has good connectivity with cloud infrastructure like Kafka. Marconi provides similar functionality implemented in Haskell to make it easy for Haskell programmers to customize it.
+
+### Scrolls 
+
+Like Scrolls, we can be selective for what is to be indexed. Scrolls can store into multiple databases. Marconi is currently focused on local DBs like SQL lite. 
+
+### Carp 
+
+## Customizing Marconi
+
+Specific applications customize Marconi by defining a custom datatype that is passed in the event stream along with functions for translating to and from the type. 
+
+Assumption: Marconi has a typeclass for the necessary functions. 
+
+The streaming component is customized with a filter function that translates blocks into user-defined types that are wrapped in the streaming event type.
+
+TODO: Decide/document the filter function and the API for customization.
```

Comment:
The current design uses the `streaming` library so the user can make use of its many combinators.

Reply:
Comment incorporated.